crawl-ref/source/dgn-overview.cc

```diff
@@ -1094,3 +1064,164 @@ bool connected_branch_can_exist(branch_type br)
+    // the current implementation shows all known branches.
+    vector<branch_type> brs;
+    for (branch_iterator it; it; ++it)
+        if (is_known_branch_id((**it).id))
+            brs.push_back(it->id);
+
+    while (true)
+    {
+        // Make the little overview
+        // (D) Dungeon        (T) Temple         (L) Lair           etc.
+        // at most 4 branches on 1 line
+        clear_messages();
+        int linec = 0;
+        string line;
+        for (branch_type br : brs)
+        {
+            if (linec == 4)
+            {
+                linec = 0;
+                mpr(line);
+                line = "";
+            }
+            line += make_stringf("(%c) %-14s ",
+                                 branches[br].travel_shortcut,
+                                 branches[br].shortname);
+            ++linec;
+        }
+        if (!line.empty())
+            mpr(line);
+
+        // tells you some special keys, but not all
+        mprf(MSGCH_PROMPT, "Annotate which branch? (. - %s, ? -help)",
+             lid.describe(false, true).c_str());
+
+        int keyin = get_ch();
+        switch (keyin)
+        {
+        CASE_ESCAPE
+            return ID_CANCEL;
+        case '?':
+            return ID_HELP;
+        case '\n': case '\r': case '.':
+            return ID_HERE;
+        case '<':
+            return ID_UP;
+        case '>':
+            return ID_DOWN;
+        case CONTROL('P'):
+            {
+                const branch_type parent = parent_branch(lid.branch);
+                if (parent < NUM_BRANCHES)
+                    return parent;
+            }
+            break;
+        default:
+            // Is this a branch hotkey?
+            for (branch_type br : brs)
+            {
+                if (toupper_safe(keyin) == branches[br].travel_shortcut)
+                    return br;
+            }
+            // Otherwise cancel
+            return ID_CANCEL;
+        }
+    }
+}
+
+void do_annotate()
+{
+    level_id lid  = level_id::current();
+    int branch = _prompt_annotate_branch(lid);
+    int depth;
+
+    if (branch == ID_HELP) // XXX: this help message could be improved.
+    {
+        clear_messages();
+        mpr("press the letter corresponding to the branch of the level you want to annotate, or");
+        mpr("? for help, . or enter to annotate the level you're currently in, < and > to annotate");
+        mpr("the levels upstairs and downstairs respectively, and control-P for the parent branch");
+        do_annotate();
+        return;
+    }
+
+    if (branch == ID_CANCEL)
+    {
+        canned_msg(MSG_OK);
+        return;
+    }
+
+    if (branch == ID_HERE)
+    {
+        annotate_level(lid);
+        return;
+    }
+
+    if (branch == ID_UP)
+    {
+        lid = find_up_level(lid);
+        annotate_level(lid);
+        return;
+    }
+
+    if (branch == ID_DOWN)
+    {
+        lid = find_down_level(lid);
+        annotate_level(lid);
+        return;
+    }
```

Comment:
having these keybinds isn't very useful because 1) they only work before you've selected a branch, unlike in interlevel travel, where they only work *after* you've selected a branch, and 2) you can't press > and < multiple times 

the travel code uses _travel_depth_keyfilter() to accomplish this

Reply:
currently, when you're standing on a stair, you can annotate the other side with > , I think. There is a slightly obnoxious menu that you have to press through even if you want to annotate the current level, but overall I like this functionality. the < and > shortcuts are intended to keep the old functionality. This will also be helpful if anyone is attached to the current functionality.

I don't care much for control-p as I have implemented it. I just thought, maybe someone will find it useful. Probably it would be more useful if it just annotated the branch entrance (for tournament challenges etc).